File: staGledas.Model/Models/Pratitelji.cs
using System;

namespace staGledas.Model.Models
{
    public class Pratitelji
    {
        public int Id { get; set; }
        public int KorisnikId { get; set; }
        public int PratiteljId { get; set; }
        public DateTime DatumPracenja { get; set; }
        public virtual Korisnici? Korisnik { get; set; }
        public virtual Korisnici? Pratitelj { get; set; }
    }
}

File: staGledas-UI/stagledas_mobile/lib/models/pratitelji.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:stagledas_mobile/models/korisnik.dart';

part 'pratitelji.g.dart';

@JsonSerializable()
class Pratitelji {
  int? id;
  int? korisnikId;
  int? pratiteljId;
  DateTime? datumPracenja;
  Korisnik? korisnik;
  Korisnik? pratitelj;

  Pratitelji({
    this.id,
    this.korisnikId,
    this.pratiteljId,
    this.datumPracenja,
    this.korisnik,
    this.pratitelj,
  });

  factory Pratitelji.fromJson(Map<String, dynamic> json) =>
      _$PratiteljiFromJson(json);

  Map<String, dynamic> toJson() => _$PratiteljiToJson(this);
}
